partiql-parser/src/result.rs

```diff
@@ -190,6 +190,10 @@ pub enum ParserError {
     /// Indicates that there is a problem with run-time violation of some API.
     #[error("Invalid Argument: {message}")]
     InvalidArgument { message: String },
+
+    /// Indicates that there is an internal error that isn't SyntaxError nor InvalidArgument.
```

Comment:
Minor suggested edit for the doc comment:

```suggestion
    /// Indicates that there is an internal error that was not due to user input or API violation.
```